From 63bdfa7e30c9555ea08fc9ae610c9111ae07f30d Mon Sep 17 00:00:00 2001
From: Yang Yingliang <yangyingliang@huawei.com>
Date: Thu, 3 Aug 2023 19:36:54 +0800
Subject: [PATCH 02/15] FROMGIT: regulator: max77857: fix build error in
 max77857-regulator.c

When using low verion gcc(7.5) to build the max77857-regulator driver,
got the following error:

drivers/regulator/max77857-regulator.c:312:16: error: initializer element is not constant
  .ramp_delay = max77857_ramp_table[0][0],

To fix this by introducing a macro RAMAP_DELAY_INIT_VAL to define the
value of max77857_ramp_table[0[0].

Fixes: af71cccadece ("regulator: max77857: Add ADI MAX77857/59/MAX77831 Regulator Support")
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Link: https://lore.kernel.org/r/20230803113654.818640-1-yangyingliang@huawei.com
Signed-off-by: Mark Brown <broonie@kernel.org>
Link: https://git.kernel.org/broonie/regulator/c/93083725e1ed86c07ea9846bb573b9340985853f
---
 drivers/regulator/max77857-regulator.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/regulator/max77857-regulator.c b/drivers/regulator/max77857-regulator.c
index f2141b5be780..44a510547318 100644
--- a/drivers/regulator/max77857-regulator.c
+++ b/drivers/regulator/max77857-regulator.c
@@ -293,8 +293,10 @@ static const unsigned int max77857_switch_freq[] = {
 	1200000, 1500000, 1800000, 2100000
 };
 
+#define RAMAP_DELAY_INIT_VAL 1333
+
 static const unsigned int max77857_ramp_table[2][4] = {
-	{ 1333, 667, 333, 227 }, /* when switch freq is 1.8MHz or 2.1MHz */
+	{ RAMAP_DELAY_INIT_VAL, 667, 333, 227 }, /* when switch freq is 1.8MHz or 2.1MHz */
 	{ 1166, 667, 333, 167 }, /* when switch freq is 1.2MHz or 1.5MHz */
 };
 
@@ -309,7 +311,7 @@ static struct regulator_desc max77857_regulator_desc = {
 	.n_ramp_values = ARRAY_SIZE(max77857_ramp_table[0]),
 	.ramp_reg = MAX77857_REG_CONT3,
 	.ramp_mask = GENMASK(1, 0),
-	.ramp_delay = max77857_ramp_table[0][0],
+	.ramp_delay = RAMAP_DELAY_INIT_VAL,
 	.owner = THIS_MODULE,
 };
 
-- 
2.41.0

